torch/csrc/jit/passes/peephole.cpp

```diff
@@ -303,6 +303,15 @@ struct PeepholeOptimizeImpl {
               output->debugName());
           node->output()->replaceAllUsesWith(output);
         }
+      } else if (node->kind() == prim::profile) {
+        auto inp = node->input(0)->node();
+        // we remove back-to-back prim::profiles if a consumer
+        // immediately follows a producer and their profiled types are equal
+        if (inp->kind() == prim::profile && node->prev() == inp &&
+            inp->ty(attr::profiled_type) == node->ty(attr::profiled_type)) {
+          node->output()->replaceAllUsesWith(inp->output());
+          it.destroyCurrent();
```

Comment:
This is the only use of `destroyCurrent` in the file - I don't think you're supposed to `destroyCurrent` with the iterator because the next node will get skipped with `++it`. You can just call replaceAllUsesWith, we run DCE at the end of the pass.

Reply:
i was wrong here - disregard lol. destroyCurrent sets the iterator to the reverse node, to properly handle this...